tokio/src/runtime/builder.rs

```diff
@@ -65,6 +67,8 @@ pub struct Builder {
     pub(super) before_stop: Option<Callback>,
 }
 
+pub(crate) type ThreadNameFn = Arc<Box<dyn Fn() -> String + Send + Sync + 'static>>;
```

Comment:
It's a shame that we have to double-box this to make it work with `loom`...but, this isn't really a hot path, so it's probably fine for now. It would be good to open an issue against `loom` about allowing `?Sized` `Arc` type parameters, though...

Reply:
Alternatively, I don't think it's super important for the thread name function to participate in `loom` tests...I think it would be fine to just always use `std::sync::Arc` here.